### PR TITLE
Add Sphinx documentation

### PR DESCRIPTION
**Pull request by mcgibbon**
## Purpose

This PR adds Sphinx documentation, and infrastructure to automatically build and show the documentation on ReadTheDocs.

## Requirements changes:

- Added requirements for documentation building to a new `requirements_docs.txt`, and updated constraints.txt with these requirements
- Deleted requirements_wrapper.txt, as the repository currently does not support wrapper builds

## Infrastructure changes:

- CONTRIBUTING.md and CONTRIBUTORS.md are moved to the top level
- added .readthedocs.yaml configuration for readthedocs builds
- Added `make docs` and `make servedocs` directives at the top level
- Added documentation under the `docs` directory, only a barebones template in this PR to be filled out in later PRs
- Deleted `fv3core/.jenkins/actions/lint.sh` and removed it from the "install fv3core virtual environment" workflow, since this is independently tested by blocking CircleCI tests
- Updated fv3core to refer to top level constraints.txt, using a symlink
- Moved `requirements_lint.txt` from fv3core to the top level

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate
- [x] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [x] For each public change and fix in `pace-util`, HISTORY has been updated
- [x] Unit tests are added or updated for non-stencil code changes

### CODE REVIEW

**Comment by mcgibbon** [comment]
Docs build off this branch and are viewable at https://pace-all.readthedocs.io/en/latest/. We should discuss the name, `pace` is taken. For now I used `pace-all` since this might be a reasonable choice for a meta-package that installs all our packages.

After the PR is merged, the readthedocs config should be updated to build off the main branch.

File: .readthedocs.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+# Set the version of Python and other tools you might need
+build:
+  os: ubuntu-20.04
+  tools:
+    python: "3.8"
+    # You can also specify other tool versions:
+    # nodejs: "16"
+    # rust: "1.55"
+    # golang: "1.17"
```
Comment (FlorianDeconinck) [inline .readthedocs.yaml:16]:
```suggestion
```

File: .readthedocs.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+# Build documentation in the docs/ directory with Sphinx
+sphinx:
+   configuration: docs/conf.py
+
+# If using Sphinx, optionally build your docs in additional formats such as PDF
```
Comment (FlorianDeconinck) [inline .readthedocs.yaml:22]: That is nice to be able to do that for offline work

**Review by FlorianDeconinck — APPROVED** [review]
LGTM, minor clean up.

Should probably enable PDF, it is useful for offline work

File: .readthedocs.yaml (hunk 2 above)
Comment (mcgibbon) [reply .readthedocs.yaml:18]: `.readthedocs.yaml` is specifically to configure online builds on readthedocs. For offline work, you can always build PDF documentation offline with `make latexpdf` inside the `docs` folder, or with `make -C docs latexpdf`.

Did you want pdf automatic builds, or is the make target fine?

Comment (FlorianDeconinck) [reply .readthedocs.yaml:18]: Make target is fine, I hadn't consider this was the readthedocs.yaml

**Pull request merged**